src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1161,7 +1161,28 @@ class WorkspaceService(protected val userInfo: UserInfo, dataSource: SlickDataSo
 
   def workflowQueueStatus() = {
     dataSource.inTransaction { dataAccess =>
-      dataAccess.workflowQuery.countWorkflowsByQueueStatus.map(RequestComplete(StatusCodes.OK, _))
+      dataAccess.workflowQuery.countWorkflowsByQueueStatus.flatMap { statusMap =>
+
+        // determine the current size of the workflow queue
+        val numQueued = statusMap.getOrElse(WorkflowStatuses.Queued.toString, 0)
+
+        numQueued match {
+          // if the queue is nonzero, ask the audit_workflow_status table for an estimate
+          case x:Int if x > 0 => {
+            dataAccess.workflowAuditStatusQuery.queueTimeMostRecentSubmittedWorkflow map { estimate =>
+              RequestComplete(StatusCodes.OK, WorkflowQueueStatusResponse(estimate, statusMap))
+            }
+          }
+          // if the queue size is 0 (or unreadable), return 0ms. we jump through some syntactic hoops
```

Comment:
I'm not super happy with the syntax here, but I had a hell of a time getting it to work any other way. Feels like there should be a better answer.